projects/packages/publicize/src/jetpack-social-settings/class-dismissed-notices.php

```diff
@@ -62,8 +55,8 @@ public function update_dismissed_notices( $updated, $name, $value ) {
 			return $updated;
 		}
 
-		$notice            = $value['notice'];
-		$reappearance_time = $value['reappearance_time'];
+		$notice            = key( $value );
```

Comment:
Am I correct in saying that `$updated` is `true` or `false`, `$name` is the option name and `$value` is ['instagram' => 0]?

If that's the case, `$notice` would be `instagram` .

`reappearance_time` would be `$value['notice']` which is 0.

I am bit confused by this change, could you please elaborate why we would need this?

Reply:
Sure!

The first part of the change (https://github.com/Automattic/jetpack/pull/34544/commits/f7eb8bc35e764c2323077870717dad627101a459#diff-b9af7ea1c5d7705e15c3aa19241ee6e3f3b8090af8977a9951def8e88907d2d9R31) is to fix the schema, as the previous one would describe our format poorly. We want the option to store an array with key->value pairs, where the key is the name of the notice, and the value is a numeric value, which is the reappearance time.

The second part (https://github.com/Automattic/jetpack/pull/34544/commits/f7eb8bc35e764c2323077870717dad627101a459#diff-5f1e74f983a6c97e6afd793e071e2b976f2f14616174253e77739c59e07de12eR63) was to modify the UI to send the data in this format. That means that for Instagram, it would send `jerpack_social_dismissed_notices: { Instagram: 0}`, which means that we want to update the registered setting `jerpack_social_dismissed_notices`, with the value of `{ Instagram: 0}`.

The third part of the change is where your comment lies, where value would be as you said, and the notice variable stores the name, and the reappearance_time is the number. Then we add it to the array `		$dismissed_notices[ $notice ] = $reappearance_time;` and update the option

Reply:
Testing it with the API console as in the testing instructions should show if it's working well